### PR TITLE
feat: hide non-global roles on user/group modal

### PR DESCRIPTION
**Pull request by gt2345**
## Description

Update "Add/Edit User/Group" modal "Role" picker to be "global role"

## Test Plan

Assign user to a group with roles, verify that roles associated with user do not show up at edit user modal.
Assign user to a workspace with some role, verify that roles associated with workspace do not show up at edit user modal.
Assign group to a workspace with some role, verify that roles associated with workspace do not show up at edit group modal.

## Checklist

- [x] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket

WEB-1010

### CODE REVIEW

File: webui/react/src/hooks/useModal/UserSettings/useModalCreateUser.tsx [hunk 1]
```diff
@@ -123,7 +119,10 @@ const ModalForm: React.FC<Props> = ({ form, user, viewOnly, roles }) => {
             </Select>
           </Form.Item>
           <Typography.Text type="secondary">
-            Note that roles inherited from user groups or workspaces cannot be removed here.
+            Groups may have additional inherited workspace roles not reflected here. &nbsp;
```
Comment (johnkim-det) [inline webui/react/src/hooks/useModal/UserSettings/useModalCreateUser.tsx:122]: I think this one should say "Users" and "global or workspace roles"?

File: webui/react/src/services/decoder.ts [hunk 2]
```diff
@@ -42,6 +42,9 @@ export const mapV1UserRole = (res: Sdk.V1RoleWithAssignments): types.UserRole =>
   const { role, groupRoleAssignments, userRoleAssignments } = res;
   return {
     fromGroup: groupRoleAssignments?.map((g) => g.groupId),
+    fromUser:
+      (userRoleAssignments?.filter((u) => !!(u.userId && u.roleAssignment.scopeCluster)) || [])
+        .length > 0,
     fromWorkspace: userRoleAssignments
```
Comment (johnkim-det) [inline webui/react/src/services/decoder.ts:48]: do we still need `fromGroup` and `fromWorkspace`?

File: webui/react/src/services/decoder.ts [hunk 3]
```diff
@@ -58,6 +61,7 @@ export const mapV1GroupRole = (res: Sdk.V1GetRolesAssignedToGroupResponse): type
     id: role.roleId,
```
Comment (johnkim-det) [inline webui/react/src/services/decoder.ts:61]: do we still need `fromWorkspace`? 

also wondering if `scopeCluster` could be renamed here to be more clear/consistent... `fromCluster`? `fromGroup`?

File: webui/react/src/types.ts [hunk 4]
```diff
@@ -870,10 +870,12 @@ export interface Permission {
 
 export interface UserRole {
   fromGroup?: number[];
+  fromUser?: boolean;
   fromWorkspace?: number[];
```
Comment (johnkim-det) [inline webui/react/src/types.ts:874]: ditto `fromGroup` and `fromWorkspace`

**Review by johnkim-det — APPROVED** [review]
Looks good! Just a potential typo on the copy for the Users modal, and some non-blocking comments on the decoder fields.

File: webui/react/src/hooks/useModal/UserSettings/useModalCreateUser.tsx (hunk 1 above)
Comment (gt2345) [reply webui/react/src/hooks/useModal/UserSettings/useModalCreateUser.tsx]: Thanks!

File: webui/react/src/types.ts (hunk 4 above)
Comment (gt2345) [reply webui/react/src/types.ts]: Unnecessary fields removed

**Pull request merged**